pkg/dogstatsd/parser_test.go

```diff
@@ -204,64 +213,46 @@ func TestParseGaugeWithEmptyHostTag(t *testing.T) {
 	assert.InEpsilon(t, 1.0, parsed.SampleRate, epsilon)
 }
 
-func TestParseGaugeWithNoHostTag(t *testing.T) {
-	defaultHostname = "test-hostname"
-	defer func() { defaultHostname = "" }()
-
-	parsed, err := parseMetricMessage([]byte("daemon:666|g|#sometag1:somevalue1,sometag2:somevalue2"), "")
-	assert.NoError(t, err)
-
-	assert.Equal(t, "daemon", parsed.Name)
-	assert.InEpsilon(t, 666.0, parsed.Value, epsilon)
-	assert.Equal(t, metrics.GaugeType, parsed.Mtype)
-	require.Equal(t, 2, len(parsed.Tags))
-	assert.Equal(t, "sometag1:somevalue1", parsed.Tags[0])
-	assert.Equal(t, "sometag2:somevalue2", parsed.Tags[1])
-	assert.Equal(t, "test-hostname", parsed.Host)
-	assert.InEpsilon(t, 1.0, parsed.SampleRate, epsilon)
-}
-
 func TestParseGaugeWithNoTags(t *testing.T) {
-	defaultHostname = "test-hostname"
-	defer func() { defaultHostname = "" }()
-
-	parsed, err := parseMetricMessage([]byte("daemon:666|g"), "")
+	parsed, err := parseMetricMessage([]byte("daemon:666|g"), "", "default-hostname")
 	assert.NoError(t, err)
 
 	assert.Equal(t, "daemon", parsed.Name)
 	assert.InEpsilon(t, 666.0, parsed.Value, epsilon)
 	assert.Equal(t, metrics.GaugeType, parsed.Mtype)
 	assert.Empty(t, parsed.Tags)
-	assert.Equal(t, "test-hostname", parsed.Host)
+	assert.Equal(t, "default-hostname", parsed.Host)
 	assert.InEpsilon(t, 1.0, parsed.SampleRate, epsilon)
 }
 
 func TestParseGaugeWithSampleRate(t *testing.T) {
-	parsed, err := parseMetricMessage([]byte("daemon:666|g|@0.21"), "")
+	parsed, err := parseMetricMessage([]byte("daemon:666|g|@0.21"), "", "default-hostname")
 
 	assert.NoError(t, err)
 
 	assert.Equal(t, "daemon", parsed.Name)
 	assert.InEpsilon(t, 666.0, parsed.Value, epsilon)
 	assert.Equal(t, metrics.GaugeType, parsed.Mtype)
 	assert.Equal(t, 0, len(parsed.Tags))
+	assert.Equal(t, "default-hostname", parsed.Host)
 	assert.InEpsilon(t, 0.21, parsed.SampleRate, epsilon)
 }
 
 func TestParseGaugeWithPoundOnly(t *testing.T) {
```

Comment:
this test demonstrates the regression when we actually do an assertion on the hostname